torchvision/ops/stochastic_depth.py

```diff
@@ -0,0 +1,56 @@
+import torch
+from torch import nn, Tensor
+
+
+def stochastic_depth(input: Tensor, mode: str, p: float, training: bool = True) -> Tensor:
+    """
+    Implements the Stochastic Depth from `"Deep Networks with Stochastic Depth"
+    <https://arxiv.org/abs/1603.09382>`_ used for randomly dropping residual
+    branches of residual architectures.
+
+    Args:
+        input (Tensor[N, ...]): The input tensor or arbitrary dimensions with the first one
+                    being its batch i.e. a batch with ``N`` rows.
+        mode (str): ``"batch"`` or ``"row"``.
+                    ``"batch"`` randomly zeroes the entire input, ``"row"`` zeroes
+                    randomly selected rows from the batch.
```

Comment:
Do I understand correctly that with "batch" the zero-ing is done element-wise, e.g. a row can have zero and non-zero elements, whereas with "row", some rows would be entirely zeroed?

Reply:
As a follow-up I'm wondering if there could be a more descriptive name instead of "row". I don't know in which context this would be used but for example if we know the input is always a batch of feature maps, this could be "featuremap" instead or something like that?

Again, depending on the context in which this is used, this comment may not make sense. 

Context: I'm unfamiliar with the use of the term "row" in a non-2D array, but perhaps it's a common generalization?

Reply:
I also don't like the name. That's the best we came up with Brian and Francisco. If you find better names, happy to replace them. 

Currently batch drops the entire batch while row drops random records of the batch.